File: 101/exercise_rework/easy_1/reverse_it2.rb
# reverse_it2.rb

# Understanding:
# Write a method
# Input, a single string
#   - Words in the string are separated by spaces
#   - If a word is five letters or more reverse it
#   
# Output
#   - A string with all words (5 or more letter ones reverse)

def reverse_words(str)
  str_arr = str.split.map do |word|
    if word.length >= 5
      word.reverse
    else
      word
    end
  end
  str_arr.join(' ')
end

# Test Cases
puts reverse_words('Professional')          # => lanoisseforP
puts reverse_words('Walk around the block') # => Walk dnuora the kcolb
puts reverse_words('Launch School')         # => hcnuaL loohcS

File: 210/js_small_problems/easy_5/reverse_it2.js
// reverse_it2.js

function reverseWords(str) {
  return str.split(' ').map ( function(word) {
    return word.length > 4 ? word.split('').reverse().join('') : word
  }).join(' ');
}

console.log(reverseWords('Professional'));             // "lanoisseforP"
console.log(reverseWords('Walk around the block'));    // "Walk dnuora the kcolb"
console.log(reverseWords('Launch School'));            // "hcnuaL loohcS"